src/pathogena/util.py

```diff
@@ -271,48 +277,124 @@ def hash_file(file_path: Path) -> str:
     return hasher.hexdigest()
 
 
+chunk_size = int(os.getenv("NEXT_PUBLIC_CHUNK_SIZE", 5000000))  # 5000000 = 5 mb
+max_concurrent_chunks = int(os.getenv("MAX_CONCURRENT_CHUNKS", 5))
+
+
+def chunk_file(file_path: Path, chunk_size: int) -> Generator[bytes, None, None]:
+    """Splits the file into chunks of the specified size.
+
+    Args:
+        file_path (Path): Path to file being uploded
+        chunk_size (int): Size of file chunks, defaults to 5 mb
+
+    Yields:
+        Generator[bytes,None, None]: generator for getting file chunk
+    """
+    with open(file_path, "rb") as file:
+        while chunk := file.read(chunk_size):
+            yield chunk
+
+
 @retry(wait=wait_random_exponential(multiplier=2, max=60), stop=stop_after_attempt(10))
-def upload_file(
-    sample_id: int,
-    file_path: Path,
+def upload_chunk(
```

Comment:
i think it is better to align client implementation with admin-app so that we use same apis and same flow in both of them. not just upload_chunk but also following steps via upload-api

1. creating a sample
2. create an upload session
3. start upload
4. upload chunks
5. end upload
6. end upload session